src/CashuWallet.ts

```diff
@@ -94,22 +93,18 @@ class CashuWallet {
 		const { payload, amount1BlindedMessages, amount2BlindedMessages } =
 			await this.createSplitPayload(0, amount, token[0]?.proofs);
 		const { fst, snd } = await this.mint.split(payload);
-		const proofs1 = fst
-			? dhke.constructProofs(
-					fst,
-					amount1BlindedMessages.rs,
-					amount1BlindedMessages.secrets,
-					this.keys
-			  )
-			: [];
-		const proofs2 = snd
-			? dhke.constructProofs(
-					snd,
-					amount2BlindedMessages.rs,
-					amount2BlindedMessages.secrets,
-					this.keys
-			  )
-			: [];
+		const proofs1 = dhke.constructProofs(
+			fst,
+			amount1BlindedMessages.rs,
+			amount1BlindedMessages.secrets,
+			this.keys
+		);
+		const proofs2 = dhke.constructProofs(
+			snd,
+			amount2BlindedMessages.rs,
+			amount2BlindedMessages.secrets,
+			this.keys
+		);
```

Comment:
huh? why did we have that? did the behaviour not change? I guess we shouldn't receive undefined objects as responses, so this solution makes more senes anyway.

Reply:
fst and snd could be undefined one case would be our 'could not verify proofs' bug. But now error handling catches this  case.